### PR TITLE
LPS-170919 Created task update action

### PR DESCRIPTION
**Pull request by rafaelfs98**
[LPS-170919](https://issues.liferay.com/browse/LPS-170919)

### CODE REVIEW

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/services/rest/TestrayTask.ts [hunk 1]
```diff
@@ -64,27 +65,50 @@ class TestrayTaskImpl extends Rest<TaskForm, TestrayTask, NestedObjectOptions> {
...
-	}
+		response = testrayTaskUsersImpl.transformDataFromList(
+			response as APIResponse<TestrayTaskUser>
+		);
 
-	public async assignTo(task: TestrayTask, userId: number) {
-		const taskUserIds = await this.getTaskUserIds(task.id);
+		const currentTaskUserIds = (userIds || []) as number[];
+		const taskUsers = response.items;
+		const taskUserIds = taskUsers.map(({user}) => user?.id as number);
```
Comment (welitonpe) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/services/rest/TestrayTask.ts:79]: Try to maintain a patter

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/services/rest/TestrayTask.ts [hunk 2]
```diff
@@ -64,27 +65,50 @@ class TestrayTaskImpl extends Rest<TaskForm, TestrayTask, NestedObjectOptions> {
...
+		}
+	}
+
+	public async assignTo(task: TestrayTask, userIds: number[]) {
+		await this.update(task.id, {
 			dueStatus: TaskStatuses.IN_ANALYSIS,
-			name: task.name,
+			name: task.name as string,
 		});
+		await this.assignUsers(task.id, userIds);
```
Comment (welitonpe) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/services/rest/TestrayTask.ts:111]: Try to maintain the patter

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/services/rest/types.ts [hunk 3]
```diff
@@ -311,6 +311,14 @@ export type TestrayTask = {
 	subtaskScoreCompleted: string;
 	subtaskScoreIncomplete: string;
 };
```
Comment (welitonpe) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/services/rest/types.ts:313]: Add a space

**Review by welitonpe — COMMENTED** [review]
Looks good but need to maintain a pattern :+1:

**Review by samuelleolopes — APPROVED** [review]
Good Job! :fire:

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Testflow/TaskHeaderActions.tsx [hunk 4]
```diff
@@ -13,37 +13,63 @@
...
+
+	const [modalType, setModalType] = useState('assign-users');
+	const [users, setUsers] = useState<number[]>([]);
+	const {modal} = useFormModal({
+		onSave: (user: number[]) => {
+			testrayTaskImpl
+				.assignTo(testrayTask, user)
+				.then(mutateTask)
+				.then(mutateTaskUsers);
+		},
```
[the remark this answers is not in the document]
Comment (rafaelfs98) [reply modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Testflow/TaskHeaderActions.tsx:52]: Great suggestion thanks my friend

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Testflow/TestflowForm.tsx [hunk 5]
```diff
@@ -46,9 +60,17 @@ const TestflowForm = () => {
 	const {modal} = useFormModal({
 		onSave: setUsers,
 	});
-	const {buildId} = useParams();
+	const {buildId, taskId} = useParams();
 	const {actions} = useTestFlowAssign({setUsers});
```
[the remark this answers is not in the document]
Comment (rafaelfs98) [reply modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Testflow/TestflowForm.tsx:65]: Great suggestion my friend, I left the const hooks together

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/services/rest/TestrayTask.ts (hunk 2 above)
Comment (rafaelfs98) [reply modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/services/rest/TestrayTask.ts:111]: Great suggestion my friend thanks

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/services/rest/types.ts (hunk 3 above)
Comment (rafaelfs98) [reply modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/services/rest/types.ts:313]: Great suggestion my friend thanks

**Review by caiosouza15 — APPROVED** [review]
Nice work :fire:

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Testflow/TaskHeaderActions.tsx [hunk 6]
```diff
@@ -13,37 +13,63 @@
...
+	TestrayTaskUser,
+	testrayTaskImpl,
+} from '../../services/rest';
+import TestflowAssignUserModal from './modal';
 
-type TaskHeaderActionsProps = {
-	TestrayTask: TestrayTask;
-	mutateTask: KeyedMutator<TestrayTask>;
+type OutletContext = {
+	mutateTask: KeyedMutator<any>;
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Testflow/TaskHeaderActions.tsx:30]: Missing the type

```suggestion
	mutateTask: KeyedMutator<TestrayTask>;
```

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Testflow/TaskHeaderActions.tsx [hunk 7]
```diff
@@ -13,37 +13,63 @@
...
+	} = useOutletContext<OutletContext>();
+
+	const [modalType, setModalType] = useState('assign-users');
+	const [users, setUsers] = useState<number[]>([]);
+	const {modal} = useFormModal({
+		onSave: (user: number[]) => {
+			testrayTaskImpl
+				.assignTo(testrayTask, user)
+				.then(mutateTask)
+				.then(mutateTaskUsers);
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Testflow/TaskHeaderActions.tsx:51]:
```suggestion
		onSave: (userIds: number[]) => {
			testrayTaskImpl
				.assignTo(testrayTask, userIds)
				.then(mutateTask)
				.then(mutateTaskUsers);
```

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Testflow/TaskHeaderActions.tsx [hunk 8]
```diff
@@ -59,12 +85,16 @@ const TaskHeaderActions: React.FC<TaskHeaderActionsProps> = ({
 				<ClayButton
 					displayType="secondary"
 					onClick={() =>
-						testrayTaskImpl.abandon(TestrayTask).then(mutateTask)
+						testrayTaskImpl
+							.abandon(testrayTask)
+							.then(mutateTask as any)
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Testflow/TaskHeaderActions.tsx:90]: This `as any` should not be here

```suggestion
							.then(mutateTask)
```

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Testflow/TaskHeaderActions.tsx (hunk 6 above)
Comment (rafaelfs98) [reply modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Testflow/TaskHeaderActions.tsx:30]: We fixed it in one place and it broke in another, I suggest we change the any when we refactor the outlet @kevenleone

**Comment by rafaelfs98** [comment]
@kevenleone

**Review by kevenleone — APPROVED** [review]
LGTM 🔥

**Comment by kevenleone** [comment]
ci:test:sf

**Comment by kevenleone** [comment]
ci:forward

**Pull request closed without merging**